add stop/delete experiment endpoint

nn-ch opened the pull request:
- currently there are 3 celery tasks that run inference, training and metrics. these are unbounded tasks which means queue is blocked if we reach max number of workers (currently set at 6 in docker-compose), that's 2 experiments.
- add a small functionality to cancel running tasks for given experiment to be able to launch new experiments
- track list of celery task ids for each experiment

feature pr: add additional way of running training and inference jobs that are infinite in its nature

nn-ch committed:
add stop/delete experiment endpoint

- track list of celery task ids and cancel running tasks to free up the space

MaxHalford commented on api/api/experiments.py:
```diff
@@ -149,9 +150,11 @@ class Experiment(sqlm.SQLModel, table=True):  # type: ignore[call-arg]
     runner: runners.Runner = sqlm.Relationship(back_populates="experiments")
     sink_id: int = sqlm.Field(foreign_key="sink.id")
     sink: sinks.Sink = sqlm.Relationship(back_populates="experiments")
+    task_ids: str | None = sqlm.Field(default=None)
```
I think a cleaner way than using `str` is to store the IDs in a list; see [here](https://github.com/tiangolo/sqlmodel/issues/178)

nn-ch replied:
thanks, i was looking for that. 
I tried `task_ids: List[str]` and variations of that but was not working.

MaxHalford commented on api/api/experiments.py:
```diff
@@ -307,6 +315,27 @@ def create_experiment(experiment: Experiment):
+        if not exp:
+            raise HTTPException(status_code=404, detail="Experiment not found")
+        for task_id in exp.task_ids.split(","):
+            RUNNER.control.revoke(task_id, terminate=True)
```
Nice!

MaxHalford commented on api/api/experiments.py:
```diff
@@ -307,6 +315,27 @@ def create_experiment(experiment: Experiment):
+
+
+@router.delete("/{experiment_id}")
+def delete_experiment(experiment_id: int):
```
Do the tasks need to be stopped if the experiment is deleted?

nn-ch replied:
yes! good catch

MaxHalford reviewed (approved):
Looks good! Just added a few remarks.

nn-ch committed:
use List instead of str for task_ids

when deleting experiment also stop the corresponding training and inference tasks from running
